src/main/scala/is/hail/expr/ir/BinaryOp.scala

```diff
@@ -108,8 +107,6 @@ object BinaryOp {
       val ll = coerce[Boolean](l)
       val rr = coerce[Boolean](r)
       op match {
-        case DoubleAmpersand() => ll && rr
-        case DoublePipe() => ll || rr
```

Comment:
A short-circuiting implementation would be:
```
ll.mux(true, rr)
```

Reply:
no, because the missingness is checked before it reaches this point, right?

Reply:
we'd want to either fix `ApplyBinaryPrimOp` to not always return `ml || mr` as the missingness, or to define a separate thing for cases like this.

Reply:
`(ll || rr)` is already a short-circuit operator in the traditional sense, but it doesn't handle missingness here.